src/services/scaleService.ts

```diff
@@ -1,21 +1,20 @@
+import {Options} from '../types/Options';
 import canvasService from './canvasService';
 import imageService from './imageService';
 
-import {Options} from '../types/Options';
+const determineScale = ({height, width}: HTMLImageElement, {maxHeight, maxWidth, scaleImageBy}: Options): number => {
```

Comment:
I'd like to refactor later to have `optionService` take in a `InputOptions` object and return an `Options` object.

```
export interface InputOptions {
    allowCrossOriginResourceSharing?: boolean;
    quality?: number;
    maxHeight?: number;
    maxWidth?: number;
    returnOriginalOnFailure?: boolean;
    scaleImageBy?: number;
}
```

```
export interface Options {
    allowCrossOriginResourceSharing: boolean;
    quality: number;
    maxHeight: number;
    maxWidth: number;
    returnOriginalOnFailure: boolean;
    scaleImageBy: number;
}
```

Hopefully one could extend the other and add or remove the required field part. This would make `scaleService` a lot cleaner as we could remove safety checks to see if `scaleImageBy`, `maxHeight`, or `maxWidth` exist.